lib/msal-core/src/utils/RequestUtils.ts

```diff
@@ -140,7 +146,33 @@ export class RequestUtils {
      */
     static validateAndGenerateState(state: string): string {
         // append GUID to user set state  or set one for the user if null
-        return !StringUtils.isEmpty(state) ? CryptoUtils.createNewGuid() + "|" + state : CryptoUtils.createNewGuid();
+        return !StringUtils.isEmpty(state) ? RequestUtils.generateLibraryState() + "|" + state : RequestUtils.generateLibraryState();
+    }
+
+    static generateLibraryState(): string {
+        const stateObject: StateObject = {
+            state: CryptoUtils.createNewGuid(),
+            iat: TimeUtils.now()
+        };
+
+        const stateString = JSON.stringify(stateObject);
+
+        return CryptoUtils.base64Encode(stateString);
+    }
+
+    static parseLibraryState(state: string): StateObject {
+        if (CryptoUtils.isGuid(state)) {
```

Comment:
if we release this there should be no need for backwards compat right? any state read should be set by the same lib?

Reply:
This would namely be necessary for a scenario where an older version of the library generates the state value and then the new version of the library tries to read that value.

Reply:
Which may not be possible in the tab suspension scenario? Not sure yet.

Reply:
interesting, that is a valid use case.
Any problem with the reverse? 
old version reading new state?

Reply:
That would not be supported.

Reply:
It should be supported right? it might just view it as a string and not decode it but the state would match?

Reply:
I suppose so? Regardless, I don't think thats a use case that needs to be supported.

Reply:
I don't think that is a use case to be supported. I would rather prefer to kill the request than process it when the library version changes.